src/ucs/datastruct/mpool_set.h

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (C) 2021, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_MPOOL_SET_H_
+#define UCS_MPOOL_SET_H_
+
+#include <ucs/datastruct/mpool.h>
+
+BEGIN_C_DECLS
+
+
+#define UCS_MPOOL_SET_SIZE     32
+#define UCS_MPOOL_SET_MAX_SIZE UCS_BIT(30)
+
+
+/* The structure represents a set of memory pools with different element sizes.
+ * Requested memory size needs to be specified when fetching element from this
+ * set. Then it is actually fetched from the memory pool with the smallest
+ * element size which can fit the requested amount of data. This approach
+ * reduces wasting of memory when mpool elements are taken for different size
+ * data. Due to performance constrains there are some restrictions for this
+ * structure:
+ * - memory pool sizes have to be power of two
+ * - maximal supported memory pool size is limited by UCS_MPOOL_SET_MAX_SIZE
+ */
+typedef struct ucs_mpool_set {
+    /* Bitmap of mpool sizes. */
+    uint32_t            bitmap;
+
+    /* Array of pointers to mpools, where index is log2 of maximal supported
+     * element size. Different elements may point to the same mpool, depending
+     * on the number of mpool sizes requested during this set initialization.
+     */
+    ucs_mpool_t         *map[UCS_MPOOL_SET_SIZE];
+
+    /* Array of mpools used by this set. */
+    ucs_mpool_t         *mpools;
```

Comment:
maybe it should be "void *data" ? since it contains both mpools and priv data
it's not used directly by INL functions right?